src/color.rs

```diff
@@ -364,18 +366,43 @@ impl<T: Primitive + 'static> From<[T; $channels]> for $ident<T> {
     }
 }
 
+impl<T: Primitive> Deref for $ident<T> {
+    type Target = $( $rgb_equiv )::* <T>;
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { mem::transmute(&self.0) }
+    }
+}
+
+impl<T: Primitive> DerefMut for $ident<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { mem::transmute(&mut self.0) }
+    }
+}
+
+impl<T: Primitive> From<$( $rgb_equiv )::* <T>> for $ident<T> {
+    fn from(value: $( $rgb_equiv )::* <T>) -> Self {
+        unsafe { mem::transmute_copy(&value) }
+    }
+}
+
+impl<T: Primitive> From<$ident<T>> for $( $rgb_equiv )::* <T> {
+    fn from(value: $ident<T>) -> Self {
+        unsafe { mem::transmute_copy(&value) }
+    }
+}
```

Comment:
If I'm not mistaken we can avoid all of the unsafety by using [`rgb::FromSlice`](https://docs.rs/rgb/0.8.25/rgb/trait.FromSlice.html) or some alternative. Particularly the `transmute_copy` is a very big and dangerous hammer that can easily be mistaken in macro code and copy and paste.

Reply:
Done. It was a little fiddly though.